### DISCUSSION TITLE
How can I use gradle to download the repo?

### DISCUSSION THREAD

**Discussion by 6-BennyLi-9** [FIRST-Tech-Challenge/FtcRobotController#1015] [Q&A]
I'd like to download:

````
    implementation 'org.firstinspires.ftc:Inspection:9.2.0'
    implementation 'org.firstinspires.ftc:Blocks:9.2.0'
    implementation 'org.firstinspires.ftc:Tfod:9.2.0'
    implementation 'org.firstinspires.ftc:RobotCore:9.2.0'
    implementation 'org.firstinspires.ftc:RobotServer:9.2.0'
    implementation 'org.firstinspires.ftc:OnBotJava:9.2.0'
    implementation 'org.firstinspires.ftc:Hardware:9.2.0'
    implementation 'org.firstinspires.ftc:FtcCommon:9.2.0'
    implementation 'org.firstinspires.ftc:Vision:9.2.0'
    implementation 'org.firstinspires.ftc:gameAssets-CenterStage:1.0.0'
    implementation 'org.tensorflow:tensorflow-lite-task-vision:0.4.3'
    implementation 'org.tensorflow:tensorflow-lite:2.12.0'
    implementation 'androidx.appcompat:appcompat:1.2.0'
````
Into my Java Project. But I'm stuck here because I don't know the correct URL, So what URL should I use?

**Comment by cmacfarl** [comment]
So let's back up a bit.

This GitHub repo is designed to be loaded into Android Studio and built and deployed using that specific tool.

If, using Android Studio, from an initial launch, it brings up a "Welcome to Android Studio" dialog where there's an Open button.  If you've cloned the repo locally, and you click on Open, and then point the "Open File or Project" dialog at the cloned repo directory, when Android Studio opens the project, it will download all the libraries, often referred to as artifacts, from Maven Central for you.  You don't need a url, because Android Studio will do all the work for you to download and cache these libraries.

If you still want to know where they are stored on Maven Central, you can go here: https://central.sonatype.com/search?q=org.firstinspires.ftc

**Reply by 6-BennyLi-9** [reply]
Well, thank you for providing a detailed answer to my question. But what I realy want is not coloning the github repo. I want to use gradle to import this moudle into my java project.

You might see that there's a ```FTC-robot-integtated-client``` repo in my accont. But now I'm doing a new project witch might remote later.

My java project is a new project created by IDEA. I used the gradle maven and I already find the way to import the ```road-runner``` into my project, but I don't know how can I complete importing this(ftc robot controller) repo.

**Comment by AlecHub** [comment]
@6-BennyLi-9, in order to be competition legal, [FtcRobotController](..) has to be your root app.

**Reply by 6-BennyLi-9** [reply]
For real? Why?

**Reply by texasdiaz** [reply]
You are completely allowed to write team code in the team code area of the app, if that's what you're worried about. But a standard app and a standard interface allows us to support teams and ensure the safety of the system, among other things. This is no different than other major robotics competitions, though there are some that are extremely open.

**Reply by 6-BennyLi-9** [reply]
Witches mean I can build a java project for myself, and use it in my teamcode?

However, my question is still incomplete.
